### PR TITLE
fix: trim invoice

### LINKED ISSUE #1252: Trim form data
### Is your feature request related to a problem? Please describe

I just copy / pasted an invoice into the payment screen and got this error: 

image

Turns out, there was a whitespace at the very front of the invoice (which was not visible anymore):

image

### Describe the solution you'd like

We could this prevent this problem by trimming values before they are processed as whitespaces in front or at the end of a string have no real use. 

### Are you working on this?

No.

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Trim the invoice

### Link this PR to an issue

Fixes #1252 

### Type of change (Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Tested Manually

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
Is it possible to add a test for this?

**Comment by im-adithya** [comment]
I think we'll have to add a check for this after adding e2e tests for app screens

**Comment by escapedcat** [comment]
Is it possible to create a unit test which types in an address with spaces and the value will only be the address?
- Pro: We have a basic unit test for this component
- Contra: We're testing if `trim` works...

**Comment by im-adithya** [comment]
Not sure if we can add a unit test as we'll have to enter a value and then check... we do all that in e2e, right?

**Comment by escapedcat** [comment]
> Not sure if we can add a unit test as we'll have to enter a value and then check... we do all that in e2e, right?

Currently we do not have e2e tests for this.  
Regarding a unit-test I thought we could do i.e. `type("  abc   ")` and `expect(...toHaveValue("abc"))` after.  
Does this make sense to you?

File: src/app/screens/Send.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ function Send() {
               value={invoice}
               disabled={loading}
               onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-                setInvoice(event.target.value)
+                setInvoice(event.target.value.trim())
```
Comment (bumi) [inline src/app/screens/Send.tsx:127]:
```suggestion
                setInvoice(event.target.value.trim().replace(/^lightning:/i,""))
```

we should additionally remove a potential `lightning:` in the beginning. This is there if you copy a link.

**Review by bumi — APPROVED** [review]
I would prefer a unit test for this. 
e2e should test full integrations, but not such small units that check if a string is correctly edited.

**Comment by escapedcat** [comment]
With the added commit form bumi let's add a unit-test

File: src/app/screens/Send/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      </MemoryRouter>
+    );
+
+    expect(
+      await screen.getByLabelText("Lightning Invoice")
+    ).toBeInTheDocument();
+    await act(async () => {
+      await user.type(
+        screen.getByLabelText("Lightning Invoice"),
+        "    sampleinvoice  "
```
Comment (escapedcat) [inline src/app/screens/Send/index.test.tsx:34]: Can we add the `lightning:` here as well? Then it's done. Thanks for you work!

Comment (im-adithya) [reply src/app/screens/Send/index.test.tsx:34]: Done!

**Comment by escapedcat** [comment]
Thanks!

**Pull request merged**